### PR TITLE
Build image sets under docker

### PR DESCRIPTION
**Pull request by petemoore**
Let's build AWS/GCP images in a predefined docker image, in order to:

a) simplify the process for taskcluster team members
b) ensure preferred versions of tools are used
c) prepare the way for building image sets in automation

Hopefully one day we'll have Windows and Linux generic-worker worker pool machine images building under monopacker and deployable to GCP/AWS/Azure, but until then, this makes sure we have a consistent build environment for the way we do things currently.

In addition, this PR provides a lot more explicit guidance (_better docs_) about the process of building image sets.

### CODE REVIEW

**Comment by petemoore** [comment]
Tested and working (for me).

Feel free to make a test deployment of `generic-worker-win2016` worker type in AWS since nothing uses that.

```
$ imagesets/imageset-under-docker.sh aws update generic-worker-win2016
```

Make sure you to read the README.md first though! :-)

File: imagesets/README.md [hunk 1]
```diff
@@ -1,12 +1,146 @@
...
-subdirectory `<IMAGE_SET>`.
+1) You will need an AWS access key configured under `~/.aws` on your host, and
+   2FA should be enabled on your AWS account.
+
+2) You will need a valid git configuration under `~/.gitconfig` on your host with
+   a valid user/email, for committing changes to community-tc-config repo and the
+   taskcluster team password store.
+
+3) You will require the ssh private key for your git account to be available
+   under `~/.ssh`.
```
Comment (djmitche) [inline imagesets/README.md:16]: I'm not sure what this means -- for git-internal, or for github?

File: imagesets/imageset-under-docker.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# This script runs imageset.sh under docker.
+# To see how the docker image was created, see the docker subdirectory.
+# Parameters passed to this script will be passed on to imageset.sh.
+
+cd "$(dirname "${0}")"
+TAG="$(cat docker/TAG)"
+docker image inspect "${TAG}" >/dev/null 2>&1 || docker pull "${TAG}"
```
Comment (djmitche) [inline imagesets/imageset-under-docker.sh:9]: `docker run` will automatically pull if the image is not local.  Is there some other purpose to this line?

File: imagesets/imageset.sh [hunk 3]
```diff
@@ -51,8 +55,18 @@ function deploy {
     UUID="$(LC_CTYPE=C </dev/urandom tr -dc "a-z0-9" | head -c 20)"
     export UNIQUE_NAME="${IMAGE_SET}-${UUID}"
 
+    export TEMP_DIR="$(mktemp -d -t password-store.XXXXXXXXXX)"
+    export PASSWORD_STORE_DIR="${TEMP_DIR}/.password-store"
+    git clone ssh://gitolite3@git-internal.mozilla.org/taskcluster/secrets.git "${PASSWORD_STORE_DIR}"
```
Comment (djmitche) [inline imagesets/imageset.sh:60]: I like that this always uses the latest commit from upstream!

File: imagesets/imageset-under-docker.sh (hunk 2 above)
Comment (petemoore) [reply imagesets/imageset-under-docker.sh:9]: Ah, TIL. Thanks!

File: imagesets/README.md (hunk 1 above)
Comment (petemoore) [reply imagesets/README.md:16]: I'll reword this to make it clearer. The `~/.ssh` host directory is mounted inside the docker container in order to give the container's `git` command access to the user's `ssh` private key for pushing changes to `ssh://gitolite3@git-internal.mozilla.org/taskcluster/secrets.git`.

File: imagesets/imageset.sh (hunk 3 above)
Comment (petemoore) [reply imagesets/imageset.sh:60]: Thanks!
